pkg/function/backup_data_using_kopia_server.go

```diff
@@ -0,0 +1,248 @@
+	fingerprint,
+	username,
+	userPassphrase string,
+	tags []string,
+) (info *kopiacmd.SnapshotCreateInfo, err error) {
+	contentCacheMB, metadataCacheMB := kopiacmd.GetCacheSizeSettingsForSnapshot()
+	configFile, logDirectory := kankopia.CustomConfigFileAndLogDirectory(hostname)
+
+	cmd := kopiacmd.RepositoryConnectServerCommand(kopiacmd.RepositoryServerCommandArgs{
+		UserPassword:    userPassphrase,
+		ConfigFilePath:  configFile,
+		LogDirectory:    logDirectory,
+		CacheDirectory:  kopiacmd.DefaultCacheDirectory,
+		Hostname:        hostname,
+		ServerURL:       serverAddress,
+		Fingerprint:     fingerprint,
+		Username:        username,
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+	})
+
+	stdout, stderr, err := kube.Exec(cli, namespace, pod, container, cmd, nil)
+	format.Log(pod, container, stdout)
+	format.Log(pod, container, stderr)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to connect to Kopia Repository Server")
+	}
+
+	cmd = kopiacmd.SnapshotCreate(
+		kopiacmd.SnapshotCreateCommandArgs{
+			PathToBackup: includePath,
+			CommandArgs: &kopiacmd.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: configFile,
+				LogDirectory:   logDirectory,
+			},
+			Tags:                   tags,
+			ProgressUpdateInterval: 0,
+			Parallelism:            utils.GetEnvAsIntOrDefault(kankopia.DataStoreParallelUploadName, kankopia.DefaultDataStoreParallelUpload),
+		})
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to construct snapshot create command")
+	}
+	stdout, stderr, err = kube.Exec(cli, namespace, pod, container, cmd, nil)
+	format.Log(pod, container, stdout)
+	format.Log(pod, container, stderr)
+
+	message := "Failed to create and upload backup"
+	if err != nil {
+		if strings.Contains(err.Error(), kerrors.ErrCodeOutOfMemoryStr) {
+			message = message + ": " + kerrors.ErrOutOfMemoryStr
+		}
+		return nil, errors.Wrap(err, message)
+	}
+	// Parse logs and return snapshot IDs and stats
+	return kopiacmd.ParseSnapshotCreateOutput(stdout, stderr)
+}
+
+func hostNameAndUserPassPhraseFromRepoServer(userCreds string) (string, string, error) {
+	var userAccessMap map[string]string
+	if err := json.Unmarshal([]byte(userCreds), &userAccessMap); err != nil {
+		return "", "", errors.Wrap(err, "Failed to unmarshal User Credentials Data")
+	}
+
+	var userPassPhrase string
+	var hostName string
+	for key, val := range userAccessMap {
+		hostName = key
+		userPassPhrase = val
+	}
+
+	decodedUserPassPhrase, err := base64.StdEncoding.DecodeString(userPassPhrase)
+	if err != nil {
+		return "", "", errors.Wrap(err, "Failed to Decode User Passphrase")
+	}
+	return hostName, string(decodedUserPassPhrase), nil
+
+}
+
+func userCredentialsAndServerTLS(tp *param.TemplateParams) (string, string, error) {
+	userCredJSON, err := json.Marshal(tp.RepositoryServer.Credentials.ServerUserAccess.Data)
+	if err != nil {
+		return "", "", errors.Wrap(err, "Error Unmarshalling User Credentials Data")
```

Comment:
```suggestion
		return "", "", errors.Wrap(err, "Error marshalling User Credentials Data")
```
OR
"Error serializing User Credentials Data"